
# Problem Set 9
Holly Do

In [2]:
import random as rnd
import math as mth
import statistics as sts
import numpy as np
import scipy as sci
import scipy.linalg as la
import scipy.stats as st
import scipy.special as spp
import matplotlib.pyplot as plt
import sympy as sy
import stemgraphic as stm

import itertools as it
import operator as op

In [3]:
# Confidence interval of Z distribution
def z_CI(xbar,sigma,n,alpha):
    '''
    xbar is the estimate for mu
    sigma is the standard deviation
    alpha is the significance level
    '''
    factor = abs(st.norm.ppf(alpha/2)*(sigma/mth.sqrt(n)))
    return (xbar - factor, xbar+factor)

In [4]:
# Confidence interval of T distribution
def t_CI(xbar,sample_std,n,alpha):
    '''
    Use t scoring for CI
    '''
    factor = abs(st.t(n-1).ppf(alpha/2)*(sample_std/mth.sqrt(n))) #difference is (n-1)
    return (xbar - factor, xbar+factor)

In [5]:
### Define function for this CI
def z_CI_diff_p(p1,p2,n1,n2,alpha):
    '''
    z-score CI for difference of proportions
    '''
    pdiff = p1 - p2
    adj_std = mth.sqrt((p1*(1-p1)/n1) + (p2*(1-p2)/n2))
    factor = abs(st.norm.ppf(alpha/2)*adj_std)
    return (pdiff - factor, pdiff+factor)

# 1.1

In [6]:
prob1 = np.array([3, 3, 3, 3, 3, 7, 7, 7])

In [7]:
np.mean(prob1)

np.float64(4.5)


Method of Moments for $\theta$ :  
How many moments are needed for 1 parameter?
For distribution that has 1 parameter, we need one moment. 

Formula for moment? Formular for moment is the formular of the mean of the distribution.  
>$
   \mu = 3 \theta + 7 (1-\theta)
$

Sample mean = $4.5$

Estimate for $\theta$ 
>$
    3 \theta + 7 (1-\theta) = 4.5
$
>$
    -4 \theta +7 = 4.5
$
>$
    \theta = \dfrac{7-4.5}{4} = .625
$


# 1.2
Maximum Likelyhood Estimator for $\theta$ :
  
  
Likelihood function is the probability function of the sample collected: $\mathcal{L}(\theta) = \theta^5 (1-\theta)^3$  
Log-Likelihood $L(\theta) = 5 \log(\theta) + 3 \log(1-\theta)$

To find the max of the probablity of the product, because the log function is increasing, we can use the maximum of the log-likelihood function.
  
Set derivative = 0 (maximize likelihood)  
>$
\dfrac{dL}{d\theta} = \dfrac{5}{\theta} - \dfrac{3}{1-\theta}
$  

>$
\dfrac{5}{\theta} - \dfrac{3}{1-\theta} = 0
$

>$
5 - 5\theta - 3\theta = 0, \theta = \frac{5}{8}
$





# 2.1

Given:
* 100 samples > 50 threshold, so Central Limit Theorem says the mean is normally distributed
* Given a valid population standard deviation $\sigma$

Formula to use : Z distribution

In [8]:
## define a function for type of CI

def z_CI(xbar,sigma,n,alpha):
    '''
    xbar is the estimate for mu
    sigma is the standard deviation
    alpha is the significance level
    '''
    #find a at alpha/2 in standard normal (aka Z score), scale to standard error of the mean (population std / square root n)
    factor = abs(st.norm.ppf(alpha/2)*(sigma/mth.sqrt(n)))
    #rang (sample_mean - margin of error, sample_mean + margin of error)
    return (xbar - factor, xbar+factor)

In [9]:
#sample mean
xbar = 37.7
#pop standardd deviation 
sigma = 9.2
#sample size
n = 100
#90% confident CI meaning significance level is 10%
alpha = 1 - .9
#calculate factor and return 90% confident CI
z_CI(xbar,sigma,n,alpha)

(np.float64(36.186734663204646), np.float64(39.21326533679536))

# 2.2

In [9]:
## compute CI with 0.01 sig
alpha = .01
z_CI(xbar,sigma,n,alpha)

(np.float64(35.330237040735014), np.float64(40.06976295926499))

Since the CI is (35.33 to 40.07) and the value 35 falls inside this range, there is not significant evidence that the mean is greater than 35 at the 1% significance level.

# 3.1

Given: 
* Underlying dist is normal ==> mean is normal
* Given a polulation std $\sigma$

Formula to use : Z distribution

In [10]:
xbar = 42
sigma = 5
n = 64
alpha = 1 - .95
lo,hi = z_CI(xbar,sigma,n,alpha)
lo,hi

(np.float64(40.77502250966246), np.float64(43.22497749033754))

# 3.2

In [12]:
#population mean
mean = 40
#true distribition given population mean and population std
dist3 = st.norm(40,5)
#True probabilty of the 3.1 CI interval is the cdf of the high point minus cdf of the low point in CI range of the true distribution. 
dist3.cdf(hi) - dist3.cdf(lo)

np.float64(0.17894383251606694)

The probability that the installation time will be within the interval computed in 3.1 is 17.9%

# 4

Given:
* Small sample size
* Unknown population std and variance

Formula to use : T distribution

In [13]:
def t_CI(xbar,sample_std,n,alpha):
    '''
    Use t scoring for CI
    '''
    factor = abs(st.t(n-1).ppf(alpha/2)*(sample_std/mth.sqrt(n))) #difference is (n-1)
    return (xbar - factor, xbar+factor)

sal = np.array([30,50,70])
xbar = np.mean(sal)
ssig = np.std(sal, ddof=1) # T-dist corrected sample std - one time only - When estimating s from the sample, the sample standard deviation itself uses n−1 in its formula for variance
n = len(sal)
alpha = .1

t_CI(xbar,ssig,n,alpha)

(np.float64(16.282910783038318), np.float64(83.71708921696168))

At at a 10% level of significance, value 80 falls inside the CI range(16.29 to 83.72) so there is not sufficient evidence that the average salary is different from 80 (in $1000s).

# 5.1

Given:
* Large sample size of 200
* Proportion problem

Formula to use : Z distribution

In [14]:
#sample mean is p
phat = 24 / 200
n = 200
#significance is 1 - confidence
alpha = 1 - .96

#Because the test is a series of bernoulli trials, sample standard deviation is sqrt(p * (1-p)) 
sigma = mth.sqrt(phat*(1-phat))

lo, hi = z_CI(phat,sigma,n,alpha)
lo, hi

(np.float64(0.07280844289046505), np.float64(0.16719155710953493))

# 5.2

Given:
* Probablity of defective items is <= 0.1
* CI at 4% level of significance is (0.073 to 0.167)

The manufacturer claims that at most one in 10 items in the shipment is defective. At the 4% level of significance, the given proportion value 0.1 falls inside the CI range, thus there is not sufficient to disprove the claim or to prove that the true proportion is different than 0.1.

# 5.3

In [15]:
#sample mean is p
phat = 24 / 200
n = 200
alpha = 0.15
sigma = mth.sqrt(phat*(1-phat))

lo, hi = z_CI(phat,sigma,n,alpha)
lo, hi

(np.float64(0.086922085134124), np.float64(0.153077914865876))

Given:
* Probablity of defective items is <= 0.1
* CI at 4% level of significance is (0.087 to 0.153)

At the 15% level of significance, the given proportion value 0.1 falls inside the CI range, thus there is not sufficient to disprove the claim or to prove that the true proportion is different than 0.1.

# 6.1

Given:
* Underlying normal distribution
* Large sample size of 52
* Population standard deviation is not given 


Formula to use : T distribution

In [15]:
xbar = 0.62
sigma = 0.2
n - 52
alpha = 1-0.95

lo, hi = z_CI(xbar,sigma,n,alpha)
lo, hi

(np.float64(0.5922819235130065), np.float64(0.6477180764869935))

# 6.2

In [16]:
# The mean's std or standard error = poplation std / square root n
std = .2
n = 52
xbar_std = std/mth.sqrt(n)

# Distribution of resistance is normal with population mean 0.6 
mu = .6

# Distrution of the MEAN of resistance has mean is the same mean of 0.6 and the standard error calculated above
d = st.norm(mu,xbar_std)

# The probability that an average of 52 resistances is 0.62 ohms or higher is 1 minus the cummulative probabilty that resistances is less than 0.62
1 - d.cdf(.62)

np.float64(0.23542084099531557)

The probability that an average of 52 resistances is 0.62 ohms or higher is 23.5%

# 7

Given:
* Large sample sizes of 72 and 100
* Proportion problem

Formula to use : Z distribution

In [1]:
### Define function for this CI
def z_CI_diff_p(p1,p2,n1,n2,alpha):
    '''
    z-score CI for difference of proportions
    '''
    #differnce of two proportions
    pdiff = p1 - p2
    
    #std of difference of two proportions with variance = p *(1-p)
    adj_std = mth.sqrt((p1*(1-p1)/n1) + (p2*(1-p2)/n2))

    factor = abs(st.norm.ppf(alpha/2) * adj_std)
    return (pdiff - factor, pdiff+factor)

nA = 70
nB = 100
voteA = 42
voteB = 59
phatA = voteA/nA
phatB = voteB/nB
alpha = 1-.95
z_CI_diff_p(phatA,phatB,nA,nB,alpha)

NameError: name 'mth' is not defined

The difference in support that this candidate is between -14%  to 16% with 95% confidence.

# 8.1

Given:
* Large sample sizes of 250 and 300
* Proportion

Formula to use : Z distribution

In [18]:
lot_a = 250
def_a = 10
lot_b = 300
def_b = 18
alpha = 1 - 0.98
phat_a = 10/250
phat_b = 18/300
z_CI_diff_p(phat_a,phat_b, lot_a , lot_b ,alpha)

(np.float64(-0.06299654797362414), np.float64(0.022996547973624146))

A 98% confidence interval for the difference of proportions of defective items is (-0.063 to 0.023).

# 8.2

At a significance level α = 0.02, there is not a significant difference between the quality of the two lots because the difference of zero falls inside the CI range.

# 9.1

In [19]:
# If we have population sigma, we divide by root n to calculate margin of error 

Given:
* Large sample sizes of 900
* proportion problem

Formula to use : Z distribution

In [20]:
p_a = 0.45
p_b = 0.35
n_a = 900
n_b = 900

dif = p_a - p_b

alpha = 1-0.95

std_a = mth.sqrt( p_a * (1-p_a) )
std_b = mth.sqrt( p_b * (1-p_b) )

dif_std = mth.sqrt((p_a * (1-p_a)/n_a) + p_b * (1-p_b)/n_b)

margin_error_a = abs(st.norm.ppf(alpha/2) * std_a/mth.sqrt(n_a) )
margin_error_b = abs(st.norm.ppf(alpha/2) * std_b/mth.sqrt(n_b))

margin_error_dif = abs(st.norm.ppf(alpha/2) * dif_std)

print( margin_error_a, margin_error_b, margin_error_dif)



0.032502325696646454 0.031161441303635233 0.04502706519206852


Margin of error the news agency should report for each of the listed estimates:
45% has margin of error +- 3.25%

35% has margin error +- 3.12%

10% has margin error +- 4.5%